src/test/java/seedu/address/logic/parser/FindCommandParserTest.java

```diff
@@ -1,34 +1,124 @@
+    private List<String> priorities = new ArrayList<>();
 
     @Test
-    public void parse_emptyArg_throwsParseException() {
-        assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+    public void parse_emptyArg_failure() {
+        assertParseFailure(parser, "     ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
     }
 
     @Test
-    public void parse_validArgs_returnsFindCommand() {
-        // no leading and trailing whitespaces
+    public void parse_validArgsForName_returnsFindCommand() {
+        names.addAll(Arrays.asList("Alice", "Bob"));
         FindCommand expectedFindCommand =
-                new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList("Alice", "Bob")));
-        assertParseSuccess(parser, "Alice Bob", expectedFindCommand);
+                new FindCommand(names, addresses, priorities);
 
-        // multiple whitespaces between keywords
-        assertParseSuccess(parser, " \n Alice \n \t Bob  \t", expectedFindCommand);
+        // No leading and trailing whitespaces
+        assertParseSuccess(parser, " n/Alice n/Bob", expectedFindCommand);
+
+        // Multiple whitespaces between successive prefixes
+        assertParseSuccess(parser, " n/ \n Alice \n \t n/ Bob  \t", expectedFindCommand);
+    }
+
+    @Test
+    public void parse_invalidArgsForName_failure() {
+        // Non-alphanumeric characters
+        assertParseFailure(parser, " n/%*!",
+                String.format(Name.MESSAGE_CONSTRAINTS + "\n" + FindCommand.MESSAGE_USAGE));
+
+        // Blank input
+        assertParseFailure(parser, " n/",
+                String.format(Name.MESSAGE_CONSTRAINTS + "\n" + FindCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgsForAddress_returnsFindCommand() {
+        addresses.addAll(Arrays.asList("Clementi", "Serangoon"));
+        FindCommand expectedFindCommand = new FindCommand(names, addresses, priorities);
+
+        // No leading and trailing whitespaces
+        assertParseSuccess(parser, " a/Clementi a/Serangoon", expectedFindCommand);
+
+        // Multiple whitespaces between successive prefixes
+        assertParseSuccess(parser, " a/ \n Clementi \n  \t   a/ Serangoon \n  \t", expectedFindCommand);
+    }
+
+    @Test
+    public void parse_invalidArgsForAddress_failure() {
+        // Blank input
+        assertParseFailure(parser, " a/",
+                String.format(Address.MESSAGE_CONSTRAINTS + "\n" + FindCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_multipleArgsWithOneAddressPrefix_returnsIncorrectFindCommand() {
+        addresses.addAll(Arrays.asList("Clementi", "Serangoon"));
+        FindCommand expectedFindCommand = new FindCommand(names, addresses, priorities);
+
+        // Only use one a/ prefix for multiple address filters
+        assertIncorrectCommand(parser, " a/Clementi Serangoon", expectedFindCommand);
+    }
```

Comment:
For this test, my intention was to show that using multiple filters with only one prefix doesn't work as what was intended, i.e. there will only be one keyword to filter by rather than the intended two keywords since the a/ prefix was not specified for the second filter. So it models a case where the user might have forgotten to use the a/prefix for their subsequent filters and hence, the find command created is not what is expected.

Reply:
Yeah, that makes sense. Thanks for the clarification!